sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -842,6 +842,30 @@ def is_fast_reboot_enabled():
             fastboot_enabled = True
 
     return fastboot_enabled
+
+# Returns 'True' if the Host Tx signal(from the host towards the QSFP-DD)
+# is indicated to be good by Orchagent, 'False' otherwise
+def is_host_tx_ready(lport):
+    host_tx_ready = 'False'
+    state_db_host =  daemon_base.db_connect("STATE_DB")
+    port_tbl = swsscommon.Table(state_db_host, 'PORT_TABLE')
+    keys = port_tbl.getKeys()
+
+    if lport in keys:
+       cmd = 'sonic-db-cli STATE_DB hget "PORT_TABLE|{}" host_tx_ready'.format(lport)
```

Comment:
Pmon is global container, how this wok under multi NPU? I mean sonic-db-cli does not require to look namesapce db ?

Reply:
> Pmon is global container, how this wok under multi NPU? I mean sonic-db-cli does not require to look namesapce db ?

@prgeor updated that he would make this thing namespace aware i.e. xcvrd would subscribe/listen to all namespace's stateDB and then act accordingly. This piece yet to be added

Reply:
@sacnaik @shyam77git please check the updated code